libs/ts-rest/core/src/lib/client.ts

```diff
@@ -108,27 +108,27 @@ type DataReturnArgs<
   Without<DataReturnArgsBase<TRoute, TClientArgs>, never>
 >;
 
-export type ApiRouteResponseNoUnknownStatus<T> =
+export type ApiRouteResponseNoUnknownStatus<T extends AppRoute> =
   | {
-      [K in keyof T]: {
+      [K in keyof T['responses']]: {
         status: K;
-        body: ZodInferOrType<T[K]>;
+        body: ZodInferOrType<T['responses'][K]>;
       };
-    }[keyof T];
+    }[keyof T['responses']];
 
-export type ApiRouteResponse<T> =
-  | ApiRouteResponseNoUnknownStatus<T>
-  | {
-      status: Exclude<HTTPStatusCode, keyof T>;
-      body: unknown;
-    };
+export type ApiRouteResponse<T extends AppRoute> = T extends { strict: true }
```

Comment:
Is this breaking change acceptable?

Reply:
Unfortunately not, however, we can instead of move the conditional type logic to the `ApiResponseForRoute` type alias.

Reply:
Will do that!

Reply:
Undone this change :)